### PR TITLE
Add MonotoneCubicInterpolation doco.

### LINKED ISSUE #8244: Need documentation for MonotoneCubicInterpolation utility
### Description of the enhancement or error report

No current documentation of utility.

### Rationale for the enhancement or information for reproducing the error

Documentation is nice.

### PR DESCRIPTION
**Pull request by lindsayad**
Fixes #8244

### CODE REVIEW

File: framework/include/utils/MonotoneCubicInterpolation.h [hunk 1]
```diff
@@ -21,37 +21,84 @@
...
+ * The algorithm used is laid out in Fritsch and Carlson, SIAM J. Numer. Anal.
+ * Vol. 17(2) April 1980
+ */
 class MonotoneCubicInterpolation
 {
 public:
+  /// Empty constructor
   MonotoneCubicInterpolation();
+
+  /* Constructor, Takes two vectors of points for which to apply the fit.  One
```
Comment (jwpeterson) [inline framework/include/utils/MonotoneCubicInterpolation.h]: Doxygen docs should start with `/**`

File: framework/include/utils/MonotoneCubicInterpolation.h [hunk 2]
```diff
@@ -21,37 +21,84 @@
...
+   * should be of the independent variable while the other should be of the
+   * dependent variable. These values should have a one-to-one correspondence,
+   * e.g. the vectors must be of the same size.
+   */
   MonotoneCubicInterpolation(const std::vector<Real> & x, const std::vector<Real> & y);
 
   virtual ~MonotoneCubicInterpolation() = default;
 
+  /* Method generally used when MonotoneCubicInterpolation object was created
```
Comment (jwpeterson) [inline framework/include/utils/MonotoneCubicInterpolation.h]: Doxygen docs should start with `/**`.

**Review by jwpeterson — CHANGES_REQUESTED** [review]
Thanks for adding docs!

**Comment by lindsayad** [comment]
Mmm, I mistakenly thought `///` was also valid Doxygen syntax. Misread the docs. Will fix

**Comment by lindsayad** [comment]
Lol, I'm not even looking at the right thing.

**Comment by jwpeterson** [comment]
> Mmm, I mistakenly thought /// was also valid Doxygen syntax.

Yeah, triple-slash is also fine.

**Review by jwpeterson — APPROVED** [review]

**Pull request merged**